src/core/buffer/representation_buffer.ts

```diff
@@ -479,9 +491,12 @@ export default function RepresentationBuffer<T>({
       }
       downloadQueue = [];
       startQueue$.next(undefined); // (re-)start with an empty queue
-      return state.type === "full-buffer" ? state : {
-        type: "idle-buffer",
-        value: undefined,
+      return state.type === "full-buffer" ? {
+        type: "full-buffer" as "full-buffer",
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=216&issues=AWLjaheq-pl6AHs2a4J6&open=AWLjaheq-pl6AHs2a4J6)

[ISSUE_KEY]: AWLjaheq-pl6AHs2a4J6